### PR TITLE
Run IntelliJ inspections on Travis

### LINKED ISSUE #8366: improve Teamcity build speed
Teamcity check is great but takes about 100 mins to complete right now compared to travis build which finished in about 30 mins.
Ref : https://teamcity.jetbrains.com/viewLog.html?buildId=2490978&buildTypeId=OpenSourceProjects_Druid_InspectionsPullRequests

### PR DESCRIPTION
**Pull request by ccaominh**
Fixes #8366.

### Description

Running IntelliJ inspections currently takes about 90 minutes, but they can be run in about 30 minutes on Travis using a docker container that contains IntelliJ: https://github.com/ccaominh/intellij-inspect. Some code changes were required to make it pass the new intellij inspection Travis job (perhaps because the Travis job uses IntelliJ 2018.3.6, whereas teamcity uses IntelliJ 2017.2.4)

<hr>

This PR has:
- [x] been self-reviewed.

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/vector/GroupByVectorColumnSelector.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 {
   int getGroupingKeySize();
 
+  @SuppressWarnings("unused")  // false positive unused inspection warning for "keySize"
```
Comment (leventov) [inline processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/vector/GroupByVectorColumnSelector.java]: Could you please create an issue in youtrack.jetbrains.com and link here?

File: processing/src/main/java/org/apache/druid/segment/data/ColumnarLongs.java [hunk 2]
```diff
@@ -121,14 +122,14 @@ public boolean isNull()
         @Override
         public long getLong()
         {
-          assert NullHandling.replaceWithDefault() || !isNull();
+          Preconditions.checkState(NullHandling.replaceWithDefault() || !isNull());
```
Comment (leventov) [inline processing/src/main/java/org/apache/druid/segment/data/ColumnarLongs.java]: Why this change?

The policy regarding asserts, project-wide should be discussed in the dev mailing list, probably.

Comment (clintropolis) [reply processing/src/main/java/org/apache/druid/segment/data/ColumnarLongs.java]: This change to column gets should definitely not be done imo, the current version with `assert` will not be evaluated at all at runtime unless asserts are enabled with jvm command line flags, and isNull is an expensive check if the bit is set in the bitmap.

In fact, I've been meaning to investigate if they should be removed entirely because they are still in the bytecode and could affect hotspot function inlining which takes function size into account, since these methods will be called from hot loops, but I haven't had the chance to investigate.

e.g. using columnar doubles

with assert:
```
  public double getDouble(int);
    Code:
       0: getstatic     #13                 // Field $assertionsDisabled:Z
       3: ifne          33
       6: invokestatic  #14                 // Method org/apache/druid/common/config/NullHandling.replaceWithDefault:()Z
       9: ifne          33
      12: aload_0
      13: getfield      #2                  // Field val$nullValueBitmap:Lorg/apache/druid/collections/bitmap/ImmutableBitmap;
      16: iload_1
      17: invokeinterface #19,  2           // InterfaceMethod org/apache/druid/collections/bitmap/ImmutableBitmap.get:(I)Z
      22: ifeq          33
      25: new           #16                 // class java/lang/AssertionError
      28: dup
      29: invokespecial #17                 // Method java/lang/AssertionError."<init>":()V
      32: athrow
      33: aload_0
      34: getfield      #1                  // Field this$0:Lorg/apache/druid/segment/data/ColumnarDoubles;
      37: iload_1
      38: invokeinterface #18,  2           // InterfaceMethod org/apache/druid/segment/data/ColumnarDoubles.get:(I)D
      43: dreturn
```

no assert:
```
  public double getDouble(int);
    Code:
       0: aload_0
       1: getfield      #1                  // Field this$0:Lorg/apache/druid/segment/data/ColumnarDoubles;
       4: iload_1
       5: invokeinterface #13,  2           // InterfaceMethod org/apache/druid/segment/data/ColumnarDoubles.get:(I)D
      10: dreturn
```

File: processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/vector/GroupByVectorColumnSelector.java (hunk 1 above)
Comment (ccaominh) [reply processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/vector/GroupByVectorColumnSelector.java]: Added https://youtrack.jetbrains.com/issue/IDEA-231034

File: processing/src/main/java/org/apache/druid/segment/data/ColumnarLongs.java (hunk 2 above)
Comment (ccaominh) [reply processing/src/main/java/org/apache/druid/segment/data/ColumnarLongs.java]: I've restored the assert statement and suppressed the inspection warning instead

**Comment by ccaominh** [comment]
When I tried to use IntelliJ 2019.3, running the inspection would print several error stack traces, for example:
```
java.lang.NullPointerException
	at com.intellij.codeInspection.deadCode.UnusedDeclarationInspectionBase$2.visitElement(UnusedDeclarationInspectionBase.java:409)
	at com.intellij.codeInspection.reference.RefJavaVisitor.visitMethod(RefJavaVisitor.java:32)
	at com.intellij.codeInspection.reference.RefMethodImpl.lambda$accept$0(RefMethodImpl.java:338)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:791)
	at com.intellij.codeInspection.reference.RefMethodImpl.accept(RefMethodImpl.java:338)
	at com.intellij.codeInspection.reference.RefManagerImpl.iterate(RefManagerImpl.java:123)
	at com.intellij.codeInspection.deadCode.UnusedDeclarationInspectionBase.queryExternalUsagesRequests(UnusedDeclarationInspectionBase.java:403)
	at com.intellij.codeInspection.deadCode.UnusedDeclarationInspection.queryExternalUsagesRequests(UnusedDeclarationInspection.java:100)
	at com.intellij.codeInspection.ex.GlobalJavaInspectionContextImpl.performPostRunActivities(GlobalJavaInspectionContextImpl.java:465)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.runGlobalTools(GlobalInspectionContextImpl.java:765)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.runTools(GlobalInspectionContextImpl.java:448)
	at com.intellij.codeInspection.ex.GlobalInspectionContextBase.lambda$performInspectionsWithProgress$3(GlobalInspectionContextBase.java:244)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:585)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.codeInspection.ex.GlobalInspectionContextBase.performInspectionsWithProgress(GlobalInspectionContextBase.java:244)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.lambda$performInspectionsWithProgressAndExportResults$1(GlobalInspectionContextImpl.java:188)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:791)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.performInspectionsWithProgressAndExportResults(GlobalInspectionContextImpl.java:196)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.launchInspectionsOffline(GlobalInspectionContextImpl.java:174)
	at com.intellij.codeInspection.InspectionApplication.lambda$runUnderProgress$17(InspectionApplication.java:512)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:163)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:585)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:150)
	at com.intellij.codeInspection.InspectionApplication.runUnderProgress(InspectionApplication.java:505)
	at com.intellij.codeInspection.InspectionApplication.runAnalysis(InspectionApplication.java:320)
	at com.intellij.codeInspection.InspectionApplication.runAnalysisOnScope(InspectionApplication.java:296)
	at com.intellij.codeInspection.InspectionApplication.run(InspectionApplication.java:244)
	at com.intellij.codeInspection.InspectionApplication.lambda$execute$0(InspectionApplication.java:144)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:821)
	at com.intellij.codeInspection.InspectionApplication.execute(InspectionApplication.java:136)
	at com.intellij.codeInspection.InspectionApplication.startup(InspectionApplication.java:121)
	at com.intellij.codeInspection.InspectionMain.main(InspectionMain.java:97)
	at com.intellij.idea.ApplicationLoader$startApp$8$1.run(ApplicationLoader.kt:242)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:193)
	at com.intellij.idea.ApplicationLoader$startApp$8.run(ApplicationLoader.kt:241)
	at java.util.concurrent.CompletableFuture.uniRun(Unknown Source)
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(Unknown Source)
	at java.util.concurrent.CompletableFuture$Completion.run(Unknown Source)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:908)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:781)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:424)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:698)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:423)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)
```

I haven't determined the cause and even though the inspection analysis does seem to complete successfully, I felt it would be better to not pollute the inspection output with error stack traces. However, when I tried it, I did notice that IntelliJ 2019.3.1 does print 200+ more inspection errors (mostly "Method is not reachable from entry points.") than IntelliJ 2018.3.6.

**Comment by leventov** [comment]
@ccaominh could you please create an issue in youtrack.jetbrains.com about these stack traces and a parallel issue in Druid or ccaominh/intellij-inspect linking to YouTrack issue so that there is a trackable path forward in upgrades?

**Comment by ccaominh** [comment]
> @ccaominh could you please create an issue in youtrack.jetbrains.com about these stack traces and a parallel issue in Druid or ccaominh/intellij-inspect linking to YouTrack issue so that there is a trackable path forward in upgrades?

I've created https://youtrack.jetbrains.com/issue/IDEA-232022 and https://github.com/ccaominh/intellij-inspect/issues/8. After this PR is merged, I can make a similar tracking issue for druid.

**Comment by ccaominh** [comment]
@leventov Let me know if you're ok with using IntelliJ 2018.3.6 for now. It's possible that IntelliJ 2020.1 fixes the issue I mentioned, and when the new version is available, I'll test and update https://github.com/ccaominh/intellij-inspect and druid.

**Comment by leventov** [comment]
@ccaominh I'm good with that.

**Comment by ccaominh** [comment]
I've filed https://issues.apache.org/jira/browse/INFRA-19871 to disable TeamCity for Druid.

**Comment by himanshug** [comment]
@ccaominh   :+1:
